nh.c

```diff
@@ -64,6 +64,10 @@ __attribute__((visibility("hidden"))) void nh_failsafe_destroy(nh_failsafe_t *fs
 // used afterwards.
 __attribute__((visibility("hidden"))) void nh_failsafe_uninstall(nh_failsafe_t *fs);
 
+// nh_delete_path deletes the provided path if it exists. It performs some checks
+// to detect and avoid deleting critical system files
+__attribute__((visibility("hidden"))) bool nh_delete_path(const char *path, bool is_dir);
```

Comment:
I'd prefer to make the function at the bottom static and remove this definition (think of the sections split with comments as different files; we're only putting them together in `nh.c` for simplicity).

Reply:
Fine by me, I thought about doing that, but decided to match what you had done for the rest of the functions. Static it is.